node/subsystem/src/jaeger.rs

```diff
@@ -154,91 +154,98 @@ pub fn pov_span(pov: &PoV, span_name: impl Into<String>) -> JaegerSpan {
+	Prep(JaegerConfig),
+	/// Uninitialized, suggests wrong API usage if encountered.
+	None,
 }
 
 impl Jaeger {
-    /// Spawn the jaeger instance.
-    pub fn new(cfg: JaegerConfig) -> Self {
-        Jaeger::Prep(cfg)
-    }
-
-    /// Spawn the background task in order to send the tracing information out via udp
-    pub fn launch(self) {
-        let cfg = match self {
-            Self::Prep(cfg) => cfg,
-            _ => { panic!("Must be a jaeger instance that was not launched yet, but has pending stuff") }
-        };
-        log::info!("🐹 Collecting jaeger spans for {}", cfg.destination);
-
-        let (traces_in, mut traces_out) = mick_jaeger::init(mick_jaeger::Config {
-            service_name: format!("polkadot-{}", cfg.node_name),
-        });
-
-        let resolved = cfg.destination.socket_addrs(|| None).unwrap();
-        if resolved.is_empty() {
-            panic!("Resolving of jaeger address always succeeds.");
-        }
-        let jaeger_agent: std::net::SocketAddr = resolved[0];
-
-        // Spawn a background task that pulls span information and sends them on the network.
-        let _handle = async_std::task::spawn(async move {
-            let udp_socket = async_std::net::UdpSocket::bind("0.0.0.0:0").await.unwrap();
-
-            loop {
-                let buf = traces_out.next().await;
-                // UDP sending errors happen only either if the API is misused (in which case
-                // panicking is desirable) or in case of missing priviledge, in which case a
-                // panic is preferable in order to inform the user.
-                udp_socket.send_to(&buf, jaeger_agent).await.unwrap();
-            }
-        });
-
-        let jaeger = Self::Launched {
-            traces_in,
-        };
-
-        *INSTANCE.lock().unwrap() = jaeger;
-    }
-
-    #[inline(always)]
-    fn traces_in(&self) -> Option<&Arc<mick_jaeger::TracesIn>> {
-        match self {
-            Self::Launched {
-                traces_in,
-                ..
-            } => Some(&traces_in),
-            _ => None,
-        }
-    }
-
-    fn span(&self, hash: &Hash, span_name: impl Into<String>) -> Option<mick_jaeger::Span> {
-        if let Some(traces_in) = self.traces_in() {
-            let trace_id = {
-                let mut buf = [0u8; 16];
-                buf.copy_from_slice(&hash.as_ref()[0..16]);
-                std::num::NonZeroU128::new(u128::from_be_bytes(buf)).unwrap()
-            };
-            Some(traces_in.span(trace_id, span_name))
-        } else {
-            None
-        }
-    }
+	/// Spawn the jaeger instance.
+	pub fn new(cfg: JaegerConfig) -> Self {
+		Jaeger::Prep(cfg)
+	}
+
+	/// Spawn the background task in order to send the tracing information out via udp
+	pub fn launch(self) {
+		let cfg = match self {
+			Self::Prep(cfg) => cfg,
+			_ => { panic!("Must be a jaeger instance that was not launched yet, but has pending stuff") }
```

Comment:
What's to stop someone from calling `Jaeger::None.launch()` and bringing down the node?

Reply:
Should be resolved with last batch of changes. Now everything has errors.